### DISCUSSION TITLE
Dependent Views: RegionBehaviors vs Navigation, ISupportDataContext vs Pub/Sub

### DISCUSSION THREAD

**Discussion by YkTru** [PrismLibrary/Prism#2948] [WPF]
@brianlagunas 

Hi I'm relatively new to Prism, I looked at your "Loading dependent views" course on pluralsight, and though I think I understand overall the techniques proposed, I don't understand why Prism's Navigation API + Pub/Sub can't do something similar

I mean: 

1- Adding/Removing Views: Can't I just when focussing/clicking ViewA (in the ContentControl region) navigate automatically in the RibbonRegion to ViewA's related TabItems, while hiding/closing all unrelated/irrelevant tabs?

2- Region's content: Would the most flexible (but not necessarily practical (ie overkill)) way to structure a RibbonControl be something like:
```
<Ribbon prism:RegionManager.RegionName="RibbonTabRegion">
                <TabControl prism:RegionManager.RegionName="TabControlRegion">
                    <TabItem Header="Module 1">
                        <ContentControl prism:RegionManager.RegionName="Module1Region"/>
                    </TabItem>
                    <TabItem Header="Module 2">
                        <ContentControl prism:RegionManager.RegionName="Module2Region"/>
                    </TabItem>
                </TabControl>
</Ribbon>
```
Or (I guess) not all container should be regions, except on rare cases?

3- Sharing DataContext: I thought Pub/Sub was a good way to share indirectly different Views' datacontexts, what are ISupportDataContext main differences/advantages-disavantages?

Thank you, Im having a hard time understanding when should I use RegionBehavior/Navigation, and ISupportDataContext/PubSub

**Comment by brianlagunas** [comment]
You can implement regions and navigating to dependent views any way you want. This is a personal choice based on your architecture and app requirements.

**Comment by YkTru** [comment]
@brianlagunas 

Ok but I mean, would I have the exact same app-functionnalities/features using Navigation (vs RegionBehavior) and Pub/Sub (vs IDataContext)?

It is not clear for me after watching "Loading Dependent Views" what are the advantages of using RegionBehavior+IDataContext: Better maintainability/modularity/readability/extensibility/overall architecture design/etc. ? and/or added in-app features/functionnalities/capabilities?

I will go for the approach that "check" all of the above properties and if equivalent, I'd probably go with Navig+PubSub since I find it more intuitive so are they equivalent? Which would you chose for a large app with lots of tabs/dialogs and a ribbons?

Thank you

**Reply by brianlagunas** [reply]
I encourage you to try the various approaches and decide for yourself which one you like best.

**Comment by YkTru** [comment]
@brianlagunas

Sorry I am not being clear (I'll refocus on my main question): I just wanted to know *before trying anything* if I can *totally* reproduce the exact same app functionnalities in your course "Loading Dependent Views" using "Navig+PubSub" instead of "RegionBehavior/TabAttribute+IDataContext", which would make the two approaches *totally* interchangeable/equivalent, and if not why?

Thank you
